ta4j-core/src/main/java/org/ta4j/core/indicators/helpers/AverageIndicator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2024 Ta4j Organization & respective
+ * authors (see AUTHORS)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.helpers;
+
+import org.ta4j.core.Indicator;
+import org.ta4j.core.indicators.CachedIndicator;
+import org.ta4j.core.num.Num;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Indicator to calculate the average of a list of other indicators.
+ */
+public class AverageIndicator extends CachedIndicator<Num> {
+    private final List<Indicator<Num>> indicators;
+
+    public AverageIndicator(Indicator<Num>... indicators) {
+        super(indicators[0]);
+
+        this.indicators = Arrays.asList(indicators);
+    }
+
+    @Override
+    protected Num calculate(int index) {
+        Num value = getBarSeries().numFactory().zero();
+
+        for (Indicator<Num> indicator : indicators) {
+            value = value.plus(indicator.getValue(index));
+        }
+
+        return value.dividedBy(getBarSeries().numFactory().numOf(indicators.size()));
+    }
+
+    @Override
+    public int getUnstableBars() {
```

Comment:
`calculate` should invoke this function early on to check the current `index` against the unstable bars value. If within the unstable period we should return `NaN`, otherwise proceed normally.

Reply:
Are you sure? Unstable bars sounds like the result is potentially not accurate, but not that it should be `NaN`. If I compare this with e.g. `SMAIndicator`, this one can also calculate the moving average from `index` 0. If I would feed two `SMAIndicator`'s of e.g. `barCount` 10 and 20 into `AverageIndicator`, it could calculate the average from `index` 0, but the value would not be accurate until `index` 20.

Reply:
Why would one want to execute trades (particularly use cases where there's real money on the line) on potentially inaccurate signals? If the end user application is simply analyzing for "fun" or other low/no stakes scenarios then perhaps false positives/negatives are no big deal, but it's a very different story when real money is involved. Since there's no distinction between "play/test" and "serious/production" indicators, all indicators must be the latter by default.

The [emerging convention](https://github.com/ta4j/ta4j/issues/947) is for indicators to return `NaN` within their unstable period to clearly delineate between values that can be trusted (and tradeable) from those that are not. There's already indicators that do this so your `calculate` addition loop logic has to handle this anyway (should be a test case).

Bottom line is you'll need to do _something_ to handle the unstable period scenario. It doesn't necessarily have to be `NaN`, but that is the prevailing convention. Whatever it is, it must unambiguously preserve signal integrity.

Reply:
The conversation is interesting. As you pointed out, "all indicators must be the latter by default". It's at least not consistent yet. And as I already pointed out, the term unstable bars suggests it is unstable, not that there is no data available.

But OK, I'll implement the changes.

Reply:
_is for indicators to return NaN within their unstable period to clearly delineate between values that can be trusted (and tradeable) from those that are not_ So do we need **_isStable_** method if instability is represented by NaN?

Reply:
Yes, converging on a stability identification and propagation solution will be a high priority item on the road map. I like the `isStable` approach right now but am open to alternatives.